spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala

```diff
@@ -586,6 +589,48 @@ object DataSet {
       ImageFrame.rdd(rawData)
     }
 
+    /**
+     * Extract hadoop sequence files from an HDFS path as ImageFrame
+     * @param url sequence files folder path
+     * @param sc spark context
+     * @param partitionNum partition number, default: Engine.nodeNumber() * Engine.coreNumber()
+     * @return
+     */
+    private[bigdl] def filesToRoiImageFrame(url: String, sc: SparkContext,
+      partitionNum: Option[Int] = None): ImageFrame = {
+      val num = partitionNum.getOrElse(Engine.nodeNumber() * Engine.coreNumber())
+      val rawData = sc.sequenceFile(url, classOf[BytesWritable], classOf[BytesWritable], num)
+        .map { data =>
+          val metaBytes = new COCODeserializer(ByteBuffer.wrap(data._1.getBytes))
+          val fileName = metaBytes.getString
+          val (height, width, anno) = metaBytes.getAnnotations
+
+          val labelClasses = Tensor(anno.map(_.categoryId.toFloat), Array(anno.length))
+          val bboxes = Tensor(
+            anno.toIterator.flatMap(ann => {
+              val x1 = Math.max(0, ann.bbox1)
+              val y1 = Math.max(0, ann.bbox2)
+              val x2 = Math.min(width - 1, x1 + Math.max(0, ann.bbox3 - 1))
+              val y2 = Math.min(height - 1, y1 + Math.max(0, ann.bbox4 - 1))
+              Iterator(x1, y1, x2, y2)
+            }).toArray,
+            Array(anno.length, 4))
+          val isCrowd = Tensor(anno.map(ann => if (ann.isCrowd) 1f else 0f), Array(anno.length))
+          val masks = anno.map(ann => ann.masks)
+          require(metaBytes.getInt == COCODataset.MAGIC_NUM, "Corrupt metadata")
+
+          val bis = new ByteArrayInputStream(data._2.getBytes)
```

Comment:
`bis` is not good naming

Reply:
It was intended to represent Byte Input Stream. I will change it to another name. :)